app/controllers/CheckYourAnswersController.scala

```diff
@@ -53,11 +54,17 @@ class CheckYourAnswersController @Inject()(
   val logger: Logger = Logger(this.getClass())
 
   def onSubmit(nilReturn: Boolean): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+     sessionRepository.set(request.userAnswers.copy(submitted = true))
      Redirect(routes.ReturnsController.onPageLoad(nilReturn = nilReturn))
   }
 
   def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData).async {
-    implicit request => constructPage(request)
+    implicit request =>
```

Comment:
I think you could add checkSub here and remove the pattern matching on submitted